src/test/java/definitions/AddWidgetDefinitions.java

```diff
@@ -0,0 +1,64 @@
+package definitions;
+
+import com.report.portal.business.constants.WidgetType;
+import com.report.portal.business.steps.DashboardsSteps;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Map;
+
+import static com.report.portal.core.utils.data.RandomSequenceCreator.generateRandomInt;
+
+@Getter
+@Setter
+public class AddWidgetDefinitions {
+
+    private DashboardsSteps dashboardsSteps = new DashboardsSteps();
+    private static final String DASHBOARD_NAME = "DEMO DASHBOARD";
+    private static final String FILTER_NAME = "DEMO_FILTER";
+
+    private WidgetType widgetType;
+    private String widgetName;
+
+    protected final int RANDOM_VALUE = generateRandomInt(1000, 9999);
+
+    @Given("User is on a Dashboard Page")
+    public void userIsOnADashboardPage() {
+        dashboardsSteps.assertDashboardVisible(DASHBOARD_NAME);
+    }
+
+    @And("User have chosen a Dashboard")
```

Comment:
- Should it be ```And```? Maybe ```When```?
- Also maybe make step more re-usable and provide possibility to define dashboard name from feature file? Or rename step to something like ```User chooses pre-defined dashboard```
- Maybe create all steps in present time? Like ```User chooses a Dashboard```

Reply:
Fixed!